extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/caffeine/CaffeineCacheImpl.java

```diff
@@ -293,4 +301,38 @@ private <T> T cast(Object value) {
                     "An existing cached value type does not match the type returned by the value loading function", e);
         }
     }
+
+    @SuppressWarnings("unchecked")
+    private <K, V> Uni<V> computeWithStats(K key, Function<K, Uni<V>> valueLoader) {
+        return (Uni<V>) Uni.createFrom()
+                .completionStage(cache.asMap().compute(key,
```

Comment:
The value computation needs to be delayed until subscription time with a `Supplier<CompletionStage>` here. The current code does not compute it lazily.

Reply:
I see.